nexus/src/app/background/init.rs

```diff
@@ -2,7 +2,90 @@
+//!   background tasks.  This means:
+//!     - We should be able to tell at compile-time which code activates what
+//!       specific background tasks.
+//!     - We should be able to tell at compile-time if code is attempting to
+//!       activate a background task that doesn't exist.
+//!     - It should be hard to add an Activator for a background task that is
+//!       not wired up to that task or is wired up to a different task.
+//!
+//! Ultimately, tasks are activated via the `Driver` which keeps track of tasks
+//! by name.  So how can we have code paths in Nexus refer to tasks in a way
+//! that satisfies these goals?  A conventional approach would be to have
+//! `Driver::register()` return a handle that could be used to activate the
+//! task, but then we wouldn't have the handle available until the task was
+//! running, which is too late -- see the note above about the circular
+//! dependency during initialization.  We could make the task identifiers global
+//! constants, but this is easy to mess up: someone could remove the task
+//! without removing its constant.  Then code paths could appear to activate the
+//! task but fail at _runtime_ (rather than compile-time) because the task
+//! actually doesn't exist.
+//!
+//! Instead, we assemble the `BackgroundTasks` struct, whose fields correspond
+//! to specific tasks.  This makes it super explicit what code paths are using
+//! which tasks.  And since the Activators in the struct can be created before
+//! the tasks are created, we can create this whole struct and pass it to all
+//! the background tasks (and anybody else that wants to activate background
+//! tasks), even though the actual tasks aren't wired up yet.  Then we can wire
+//! it up behind the scenes.  If someone uses the activators ahead of time,
+//! they'll get the expected behavior: the task will be activated shortly.
+//!
+//! There remain several ways someone could get this wrong when adding or
+//! reworking background tasks:
+//!
+//! - Forgetting to put an Activator for a background task into
+//!   `BackgroundTasks`.  If you make this mistake, you won't get far because
+//!   you won't have the argument you need for `Driver::register()`.
+//! - Forgetting to wire up an Activator by passing it to `Driver::register()`.
+//!   We attempt to avoid this with an exhaustive match inside
+//!   `BackgroundTasksInitializer::start()`.  If you forget to wire something
+//!   up, rustc should report an unused variable.
+//! - Wiring the Activator up to the wrong task (e.g., by copying and pasting a
+//!   `Driver::register()` call and forgetting to update the activator
+//!   argument).  If this happens, it's likely that either one Activator gets
+//!   used more than once (which is caught with a panic only at runtime, but
+//!   it _is_ during Nexus initialization, so it should definitely be caught in
+//!   testing) or else some Activator is unused (see the previous bullet).
```

Comment:
suuuuuuper nitpicky unimportant docs nit: we use the code formatting for other type and method names in these docs, but we don't use it for `Activator` for some reason:
```suggestion
//! Instead, we assemble the `BackgroundTasks` struct, whose fields correspond
//! to specific tasks.  This makes it super explicit what code paths are using
//! which tasks.  And since the `Activator`s in the struct can be created before
//! the tasks are created, we can create this whole struct and pass it to all
//! the background tasks (and anybody else that wants to activate background
//! tasks), even though the actual tasks aren't wired up yet.  Then we can wire
//! it up behind the scenes.  If someone uses the `Activator`s ahead of time,
//! they'll get the expected behavior: the task will be activated shortly.
//!
//! There remain several ways someone could get this wrong when adding or
//! reworking background tasks:
//!
//! - Forgetting to put an `Activator` for a background task into
//!   `BackgroundTasks`.  If you make this mistake, you won't get far because
//!   you won't have the argument you need for `Driver::register()`.
//! - Forgetting to wire up an `Activator` by passing it to `Driver::register()`.
//!   We attempt to avoid this with an exhaustive match inside
//!   `BackgroundTasksInitializer::start()`.  If you forget to wire something
//!   up, rustc should report an unused variable.
//! - Wiring the `Activator` up to the wrong task (e.g., by copying and pasting a
//!   `Driver::register()` call and forgetting to update the activator
//!   argument).  If this happens, it's likely that either one `Activator` gets
//!   used more than once (which is caught with a panic only at runtime, but
//!   it _is_ during Nexus initialization, so it should definitely be caught in
//!   testing) or else some `Activator` is unused (see the previous bullet).
```